app_flutter/lib/status_grid.dart

```diff
@@ -45,29 +44,128 @@ class StatusGridContainer extends StatelessWidget {
           );
         }
 
+        final Map<String, int> taskColumnKeyIndex =
+            createTaskColumnKeyIndex(statuses);
+        final List<List<Task>> taskMatrix =
+            createTaskMatrix(statuses, taskColumnKeyIndex);
+        final List<Task> taskIconRow = createTaskIconRow(statuses, taskColumnKeyIndex);
+
         return StatusGrid(
           statuses: statuses,
+          taskMatrix: taskMatrix,
+          taskIconRow: taskIconRow,
         );
       },
     );
   }
+
+  /// A unique index for grouping [Task] from separate [CommitStatus].
+  ///
+  /// [task.stageName] and [task.name] are not unique. However, they
+  /// are unique when combined together.
+  static String _taskColumnKey(Task task) {
+    return '${task.stageName}:${task.name}';
+  }
+
+  /// A map of all [taskColumnKey] to a unique index from [List<CommitStatus>].
+  ///
+  /// Scans through all [Task] in [List<CommitStatus>] to find all unique [taskColumnKey].
+  /// This ensures a column is allocated in [createTaskMatrix].
+  ///
+  /// When a new [taskColumnKey] is found, it is inserted and the index is incremeted.
+  @visibleForTesting
+  static Map<String, int> createTaskColumnKeyIndex(List<CommitStatus> statuses) {
```

Comment:
Rather than creating a series of staticMethods that are visibleForTesting, I would create a separate `TaskMatrix` or similar class and collect all of the data/methods into that. Otherwise, this is a great improvement!

Reply:
I created `StatusGridHelper` that can manage this. Down the line we can add functions for ordering the Tasks in the matrix there. Can you take a look?